File: fm-server/src/main/java/io/zeitwert/fm/server/config/crnk/CrnkBootConfig.java
package io.zeitwert.fm.server.config.crnk;

import io.crnk.core.boot.CrnkBoot;
import io.crnk.core.queryspec.mapper.QuerySpecUrlMapper;
import io.zeitwert.dddrive.persist.util.CustomFilters;
import org.springframework.stereotype.Component;

/**
 * Custom CrnkBoot configurer that adds support for the IN filter operator.
 */
@Component
public class CrnkBootConfig implements CrnkBootConfigurer {

	@Override
	public void configure(CrnkBoot boot) {
		QuerySpecUrlMapper urlMapper = boot.getUrlMapper();
		if (urlMapper instanceof DefaultQuerySpecUrlMapper mapper) {
			mapper.addSupportedOperator(CustomFilters.IN);
		}
	}

}

File: fm-server/src/main/java/io/zeitwert/fm/server/config/crnk/CrnkBootConfig.kt
package io.zeitwert.fm.server.config.crnk

import io.crnk.core.boot.CrnkBoot
import io.zeitwert.persist.sql.ddd.util.CustomFilters
import org.springframework.stereotype.Component

/**
 * Custom CrnkBoot configurer that adds support for the IN filter operator.
 */
@Component
class CrnkBootConfig : CrnkBootConfigurer {

	override fun configure(boot: CrnkBoot) {
		val urlMapper = boot.getUrlMapper()
		if (urlMapper is DefaultQuerySpecUrlMapper) {
			urlMapper.addSupportedOperator(CustomFilters.IN)
		}
	}

}
